src/Data/Geometry/Structure/Feature.hs

```diff
@@ -338,3 +389,19 @@ foreign import javascript unsafe "$1['features']"
 {-# INLINE js_JSArrayToFC #-}
 foreign import javascript unsafe "$r = {}; $r['type'] = 'FeatureCollection'; $r['features'] = $1"
     js_JSArrayToFC :: JS.Array Feature -> FeatureCollection
+
+----------------------------------------------------------------------------------------------------
+-- GeometryInput converters
+----------------------------------------------------------------------------------------------------
+
+{-# INLINE js_GIToFC #-}
+foreign import javascript unsafe "$1['geometry']"
+    js_GIToFC :: GeometryInput -> FeatureCollection
+{-# INLINE js_FCToGI #-}
+foreign import javascript unsafe "$r = {}; $r['geometry'] = $1"
+    js_FCToGI :: FeatureCollection -> GeometryInput
+
+-- If a geometryInput is actually a featureCollection
+{-# INLINE js_ChangeGIToFC #-}
```

Comment:
In the situation when you are absolutely sure that one JSVal wrapper type can be changed to another one,
you can use function coerce from Data.Coerce module. These foreign functions are overkill :)